### PR TITLE
Make the components' issue template public so it can be easily updated/consumed/shared

### PR DESCRIPTION
**Pull request by didoo**
### :pushpin: Summary

As agreed during today's team sync, I have moved the existing issue template for new components in a more public/visible folder, so that can be used as initial seed for the "definition of done" of a component, to be used as general reference but also as place where we can copy&paste snippets for the acceptance criteria of tasks in (as suggested by @gabbyhon).

Once this is merged, we can open new PRs to update it (with new content, eg. the design part, or accordingly to the teams' decisions around certain topics). Everyone can chime in and comment, open a PR, improve it, etc.
Once stable enough, we can convert to a page in Confluence or a discussion in Github.

### :hammer_and_wrench: Detailed description

In this PR I have:
- moved the issue template to the “components” folder and renamed it `Component Checklist`

Things to do/discuss/decide
- [ ] what is called this file? (which is also the title of the document?)

***

### 👀 How to review

👉 Review by files changed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

**Review by amyrlam — APPROVED** [review]
LGTM as-is

Could consider `NEW-COMPONENT-CHECKLIST.md` if you want

File: packages/components/COMPONENT-CHECKLIST.md [hunk 1]
```diff
@@ -28,7 +21,7 @@ The engineering checklist has six parts: creating the feature branch, component
 - [ ] **component template**
   - use semantic HTML
   - the component should have a css class that is the same as the component (e.g. `hds/button` should have a class name of `hds-button` on the component, and additional CSS classes should start with this same class name.
-  - add `...attributes` unless doing so would be detrimental (e.g., a parent component and child component that both have ...attributes) 
+  - add `...attributes` unless doing so would be detrimental (e.g., a parent component and child component that both have ...attributes)
```
Comment (MelSumner) [inline packages/components/COMPONENT-CHECKLIST.md:24]:
```suggestion
  - add `...attributes` unless doing so would be detrimental (e.g., a parent element and child element in the same component that both have ...attributes)
```

File: packages/components/COMPONENT-CHECKLIST.md [hunk 2]
```diff
@@ -39,7 +32,7 @@ The engineering checklist has six parts: creating the feature branch, component
   - assertion text should match the content style of the other components.
   - goal is a terse invocation
 - [ ] **component style**
-  - create `component/COMPONENT_NAME.scss` in `app/styles` 
+  - create `component/COMPONENT_NAME.scss` in `app/styles`
   - add `@use` to `app/styles/@hashicorp/design-system-components.scss` (see existing code for precise syntax)
   - use design tokens wherever possible, comment where they are not
   - sizes should be in relative units
```
Comment (MelSumner) [inline packages/components/COMPONENT-CHECKLIST.md:38]:
```suggestion
  - sizes should be in relative units
  - use BEM for naming
```

File: packages/components/COMPONENT-CHECKLIST.md [hunk 3]
```diff
@@ -53,26 +46,26 @@ The engineering checklist has six parts: creating the feature branch, component
...
-  - [ ] Usage 
-  - [ ] Showcase 
+  - [ ] Usage
+  - [ ] Showcase
 
 ### Component Review
 
 Pre-review request checks:
 
 - [ ] make sure all tests pass (`ember s` then visit /tests; or `ember t -s`)
```
Comment (MelSumner) [inline packages/components/COMPONENT-CHECKLIST.md:56]:
```suggestion
- [ ] run `yarn lint` and fix any issues (`yarn lint:hbs --fix` will resolve most issues)
- [ ] make sure all tests pass (`ember s` then visit /tests; or `ember t -s`)
```

**Review by MelSumner — APPROVED** [review]
Added some suggestions, otherwise looks good.

File: packages/components/COMPONENT-CHECKLIST.md (hunk 2 above)
Comment (didoo) [reply packages/components/COMPONENT-CHECKLIST.md:38]: This is not yet defined. We have to be more subtle here.

**Pull request merged**